### DISCUSSION TITLE
Run NPM command hook?

### DISCUSSION THREAD

**Discussion by JrobT** [renovatebot/renovate#16700] [Request Help] [answered]
Sometimes PRs from Renovate fail on my NPM repos due to linting. Is there a way I can get Renovate to run a specific NPM script before it commits? i.e. `npm run lint`? This would save me having to do it.

**Comment by rarkins** [comment, marked as answer]
In self hosting you could, but not in the app, because we don't allow arbitrary scripts

**Reply by JrobT** [reply]
Can I just clarify what you mean by I could in self-hosting? I am self-hosting for context.

**Reply by viceice** [reply]
https://docs.renovatebot.com/configuration-options/#postupgradetasks

**Reply by JrobT** [reply]
Great - thank you 🥇
